metabase/metabase
# feat(sdk): render custom question layout when drilling down in interactive dashboard

Closes https://github.com/metabase/metabase/issues/49718

### Description

Add an ability to render custom question layouts via the `renderQuestionView` prop to the `InteractiveDashboard` component. This passes the custom view component to the ad-hoc question.

### How to verify

- Use the `renderQuestionView` on interactive dashboards, per the story.

```
const QuestionView = () => (
  <div>
    <InteractiveQuestion.Title />
    <InteractiveQuestion.QuestionVisualization />
    <div>This is a custom question layout.</div>
  </div>
)

<InteractiveDashboard questionId={1} renderQuestionView={QuestionView} />
```

- Click on any dashboard card that links to a question.
- You should be taken to the question, with the specified layout from the QuestionView component.

### Checklist
